care/emr/api/viewsets/location.py

```diff
@@ -423,3 +426,15 @@ def get_queryset(self):
         ):
             raise PermissionDenied("You do not have permission to given location")
         return FacilityLocationEncounter.objects.filter(location=location)
+
+
+def close_related_location_from_encounter(instance):
+    if instance.status in COMPLETED_CHOICES:
+        with transaction.atomic():
```

Comment:
This is a single query, there is no point in it being inside an atomic transaction.